### PR TITLE
feat: map matching disease codes to categories

### PR DESCRIPTION
**Pull request by dtroelofsprins**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/system tested
- [ ] User documentation updated
- [ ] Clean commits
- [x] Updated CHANGELOG.md

### CODE REVIEW

**Comment by dtroelofsprins** [comment]
[#9169 As directory data manager I want to automatically map computed diseases to categories](https://trello.com/c/02eQgDJk/933-9169-as-directory-data-manager-i-want-to-automatically-map-computed-diseases-to-categories)

File: src/molgenis/bbmri_eric/bbmri_client.py [hunk 1]
```diff
@@ -65,22 +65,27 @@ def __init__(self, *args, **kwargs):
...
+                               "matching" ontology code columns
         :return: an OntologyTable
         """
+        matching_attrs = matching_attrs if matching_attrs else []
+
         rows = self.get(
             entity_type_id,
             batch_size=10000,
-            attributes=f"id,{parent_attr},ontology",
+            attributes=f"id,{parent_attr},ontology,{','.join(matching_attrs)}",
```
Comment (jelmerveen) [inline src/molgenis/bbmri_eric/bbmri_client.py:84]: what happens if matching_attrs is null?

File: src/molgenis/bbmri_eric/categories.py [hunk 2]
```diff
@@ -91,8 +92,13 @@ def _map_paediatric(cls, collection: dict, categories: List[str]):
                 categories.append(Category.PAEDIATRIC_INCLUDED.value)
 
     def _map_diseases(self, collection: dict, categories: List[str]):
-        diagnoses = collection.get("diagnosis_available", [])
+        diagnoses = deepcopy(collection.get("diagnosis_available", []))
```
Comment (jelmerveen) [inline src/molgenis/bbmri_eric/categories.py:95]: you had a bug?

**Review by jelmerveen — APPROVED** [review]

File: src/molgenis/bbmri_eric/categories.py (hunk 2 above)
Comment (dtroelofsprins) [reply src/molgenis/bbmri_eric/categories.py:95]: No a new piece of code was added, where matching diagnoses were added to the variable. Without deepcopy, the actual diagnoses belonging to that collection will be affected to.

File: src/molgenis/bbmri_eric/bbmri_client.py (hunk 1 above)
Comment (dtroelofsprins) [reply src/molgenis/bbmri_eric/bbmri_client.py:84]: If matching attrs is null (or actual an empty list []). These columns won't be selected.

File: src/molgenis/bbmri_eric/bbmri_client.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from collections import defaultdict
 from dataclasses import asdict, dataclass
 from enum import Enum
-from typing import List
+from typing import List, Optional
```
[the remark this answers is not in the document]
Comment (dtroelofsprins) [reply src/molgenis/bbmri_eric/bbmri_client.py]: It should make parameters Optional, but the way it was implemented it did nothing. As it turned out that in Python 3.10 the this typing could be replaced by a union type expressions, I've replaced them all and added a correct default syntax.

**Review by jelmerveen — APPROVED** [review]

**Pull request merged**